messytables/excel.py

```diff
@@ -138,10 +138,51 @@ def properties(self):
 
 class XLSProperties(CoreProperties):
     KEYS = ['bold', 'size', 'italic', 'font_name', 'strikeout', 'underline',
-            'font_colour', 'background_colour', 'any_border', 'all_border']
+            'font_colour', 'background_colour', 'any_border', 'all_border',
+            'richtext']
     def __init__(self, cell):
-        self.xf = cell.sheet.book.xf_list[cell.xlrd_cell.xf_index]
-        self.font = cell.sheet.book.font_list[self.xf.font_index]
+        #self.xf = cell.sheet.book.xf_list[cell.xlrd_cell.xf_index]
+        #self.font = cell.sheet.book.font_list[self.xf.font_index]
+        self.cell = cell
+        self.merged = {}
+
+    @property
+    def xf(self):
+        return self.cell.sheet.book.xf_list[self.cell.xlrd_cell.xf_index]
+
+    @property
+    def font(self):
+        return self.cell.sheet.book.font_list[self.xf.font_index]
+
+    @property
+    def rich(self):
+        return self.cell.sheet.rich_text_runlist_map.get(self.cell.xlrd_pos, None)
+
+    def raw_span(self, always=False):
+        "return the bounding box of the cells it's part of."
+        row, col = self.cell.xlrd_pos
+        for box in self.cell.sheet.merged_cells:
+            rlo, rhi, clo, chi = box
+            if row >= rlo and row <= rhi and col >= clo and col <= chi:
+                return box
+        if always:
+            print self.cell.xlrd_pos, box
+            return (row, row, col, col)
+        else:
+            print self.cell.xlrd_pos, None
+            return None
+
+    @property
+    def topleft(self):
```

Comment:
Should this be called is_topleft? Perhaps not. Maybe just say in docstring that it returns a boolean as to whether topleft.